packages/components/atoms/f-link/src/components/Link.vue

```diff
@@ -1,45 +1,123 @@
 <template>
-    <div
-        :class="$style['o-link']"
-        data-test-id="link"
-    >
-        {{ copy.text }}
-    </div>
+    <a
+        :class="[
+            $style['o-link'], {
+                [$style['o-link--bold']]: isBold,
+                [$style['o-link--noDecoration']]: !hasTextDecoration,
+                [$style['o-link--full']]: isFullWidth
+            }]"
+        :data-test-id="'link-component'"
+        :aria-label="ariaLabel"
+        :target="target"
+        :rel="rel"
+        v-bind="$attrs">
+        <slot />
+    </a>
 </template>
 
 <script>
-import { globalisationServices } from '@justeat/f-services';
+import { VueGlobalisationMixin } from '@justeat/f-globalisation';
 import tenantConfigs from '../tenants';
 
 export default {
     name: 'VLink',
-    components: {},
+
+    mixins: [VueGlobalisationMixin],
+
     props: {
-        locale: {
-            type: String,
-            default: ''
+        isExternal: {
+            type: Boolean,
+            default: false
+        },
+
+        opensInNewLocation: {
+            type: Boolean,
+            default: false
+        },
+
+        isBold: {
+            type: Boolean,
+            default: false
+        },
+
+        hasTextDecoration: {
+            type: Boolean,
+            default: true
+        },
+
+        isFullWidth: {
+            type: Boolean,
+            default: false
         }
     },
-    data () {
-        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
-        const localeConfig = tenantConfigs[locale];
 
+    data () {
         return {
-            copy: { ...localeConfig }
+            tenantConfigs
         };
+    },
+
+    computed: {
+        linkText () {
+            return this.$slots.default[0].children[0].text.trim();
+        },
```

Comment:
I'd worry here that any text from subsequent children after the first would be lost.

I'd be tempted to add the aria-label content using a [locally-registered directive](https://vuejs.org/v2/guide/custom-directive.html#Hook-Functions) hooking into the `componentUpdated` event to ensure that all text from child elements would be available and using the element's [`innerText`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText) property. Happy to walk through this in a pairing session if this would be useful. @ashleynolan - any thoughts on this strategy?

Reply:
Thanks for this Ben 🙂

Reply:
Nope – I'd make sure we add some comments to the resulting code just explaining what it's doing for anyone new to this sectin.